assets/js/script_spotify.js

```diff
@@ -1,82 +1,115 @@
+      console.log(response);
+      bearerToken = response.access_token;
+      console.log("bearerToken", bearerToken);
+      get_genresArray(bearerToken);     
+    },
+    error: function (response) {
+      console.log("ERROR BEARERTOKEN", response);
+    }
+  });
+  return bearerToken;
+}
 
 // Get Genres Array
-$.ajax({
-  type: "GET",
-  url: "https://api.spotify.com/v1/recommendations/available-genre-seeds",
-  headers: {
-    'Authorization': "Bearer " + token
-  },
-  contentType: "application/json",
-  async: false,
-  dataType: "json",
-  success: function (response) {
-    console.log(token);
-    console.log(response);
-  },
-  error: function (response) {
-    console.log(token);
-    console.log(response);
-  } 
-});
+function get_genresArray(token){
+  $.ajax({
+    type: "GET",
+    url: "https://api.spotify.com/v1/recommendations/available-genre-seeds",
+    headers: {
+      'Authorization': "Bearer " + token
+    },
+    contentType: "application/json",
+    async: false,
+    dataType: "json",
+    success: function (response) {
+      console.log("genreArray token", token);
+      console.log("genreArray response", response);
+      get_Tracks(token, response);
+    },
+    error: function (response) {
+      console.log("error", token);
+      console.log("error", response);
+    } 
+  });
+}
 
 // Gets List of tracks based on selected genre
-$.ajax({
-  type: "GET",
-  url: "https://api.spotify.com/v1/recommendations",
-  headers: {
-    'Authorization': "Bearer " + token
-  },
-  data: {
-    seed_genres: "alternative" // dynamic variable set by user input
-  },
-  contentType: "application/json",
-  async: false,
-  dataType: "json",
-  success: function (response) {
-    console.log(response);
-    tracks = response.tracks;
-    console.log(tracks);
-  }
-});
+function get_Tracks(token, genresArr){
+  var chosenGenre = get_Genre(genresArr);
+  console.log("chosenGenre", chosenGenre);
+  $.ajax({
+    type: "GET",
+    url: "https://api.spotify.com/v1/recommendations",
+    headers: {
+      'Authorization': "Bearer " + token
+    },
+    data: {
+      seed_genres: chosenGenre // dynamic variable set by user input
+    },
+    contentType: "application/json",
+    async: false,
+    dataType: "json",
+    success: function (response) {
+      console.log("GET TRACKS", response);
+      tracks = response.tracks;
+      console.log("TRACKS", tracks);
+      get_TrackId(tracks);
+    },
+    error: function(response){
+      console.log("error", response);
+    }
+  });
+}
+
+//choose a genre from user's input (randomly generates for now)
+//genresArr param might not be necessary, was using it to randomly generate genres -- Hailey F.
+function get_Genre(genresArr){
+  var userInput = $("#musicSearch").children("input").val();
+  console.log("userInput", userInput);
+  //TODO: get the user's input to choose a genre
+    // I believe we were talking about using a dropdown menu on the HTML side? -- Hailey F.
+  
```

Comment:
Refer to UCF-VIRT-FSF-PT-10-2022-U-LOLC • 05-Third-Party-APIs\01-Activities\11-jQuery-UI-Widgets\Solved\assets\js\script.js activity implementing the autocomplete functionality.

Reply:
Uses the Jquery UI library, added it to the HTML doc in our branch, notified the group, waiting to see if this may cause merge conflicts -- it is done but currently not styled. I added some comments on how I refactored to get the token and genre array generated on page refresh instead of button press, otherwise it would require entering genres manually.